integ/persistence_test.go

```diff
@@ -148,13 +156,39 @@ func doTestPersistenceWorkflow(
 			SearchAttributes: []iwfidl.SearchAttribute{
 				expectedDatetimeSearchAttribute,
 			},
+			DataAttributes: []iwfidl.KeyValue{
+				expectedDataAttribute,
+			},
 			WorkflowConfigOverride:   config,
 			UseMemoForDataAttributes: ptr.Any(useMemo),
 		},
 	}
 	_, httpResp, err := reqStart.WorkflowStartRequest(wfReq).Execute()
 	panicAtHttpError(err, httpResp)
 
+	initReqQry := apiClient.DefaultApi.ApiV1WorkflowDataobjectsGetPost(context.Background())
+
+	queryResult, httpResp, err := getDataAttributes(initReqQry, wfId, expectedDataAttribute, useMemo)
+
+	retryCount := 0
+
+	// Config is only present for continueAsNew tests
+	if config != nil {
+		for {
+			if err == nil || retryCount >= 5 {
+				break
+			}
+			// Loading data to a continuedAsNew workflow might take a few seconds thus retry mechanism is needed
+			time.Sleep(time.Millisecond * 1000)
+			retryCount += 1
+			queryResult, httpResp, err = getDataAttributes(initReqQry, wfId, expectedDataAttribute, useMemo)
+		}
+	}
```

Comment:
@longquanzheng what are your thoughts about this?

Reply:
That's fine. I was thinking about the same way to workaround. 

However, ideally we can do this in api service for everyone (that will saves all the sleeping in the tests).

But it's better to do in a separate PR (need to figure out checking error types, and backoff strategy and refactoring all the tests to remove the sleep).

Could you also raise another ticket for this?

Reply:
Will do!